pkg/deploy/api/validation/validation.go

```diff
@@ -39,6 +39,9 @@ func ValidateDeploymentConfigSpec(spec deployapi.DeploymentConfigSpec) field.Err
 	if spec.Template == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("template"), ""))
 	} else {
+		originalContainerImageNames := getContainerImageNames(spec.Template)
+		defer setContainerImageNames(spec.Template, originalContainerImageNames)
```

Comment:
nothing else from me. @bparees has merge, not sure if the handleEmptyImageReferences impl changed since he lgtm'ed the first time